container-build support

jessa0 opened the pull request:
[container-build](https://github.com/geogriff/container-build) is a tool I wrote to automate building a docker image and running a command within that container, geared toward setting up reproducible build environments. This PR adds the config files necessary to run the pc and rpi3 builds using that tool, with no dependencies other than python3 and docker.

I had to change the qemu-arm binfmt_misc check to actually compile a program and try to run it as opposed to looking for a specific registration in binfmt_misc, which is probably what we actually want to test anyway.

Commit:
add auxiliary files for containerized build

Commit:
test run a cross compiled program instead of checking binfmt_misc

when building in a docker container, for example, binfmt_misc isn't already mounted by default, so the binfmt_misc check
in the Makefile fails. we actually rather be testing whether the host can run cross-compiled programs, which is what we
really want to know, especially because we don't care whether they're using qemu or something else to run them.

nolanl commented on container-build/sources.list:
```diff
@@ -0,0 +1 @@
+deb [arch=amd64] https://download.docker.com/linux/debian stretch stable
```
Why is the "[arch=amd64]" needed? How does it work when doing an arm64 build?

jessa0 replied:
good catch. that specifier shouldn't be there. the arm64 build works because this sources.list is only for installing docker-ce-cli in the top level container used to run `make`.

jessa0 replied:
fixed in 5c91fcd

Commit:
remove unnecessary arch specifier in container-build sources.list

this would prevent developers from building the project on non-amd64 architectures.

nolanl reviewed (commented):
Would be nice to vendor container-build in scripts, and make it #! executable. Then we could make container-build the default suggested build method.

Commit:
vendor container-build

upstream revision 2f331af2b3fd6bf865d015670257ac3bd7ac5957

jessa0 commented:
> Would be nice to vendor container-build in scripts, and make it #! executable. Then we could make container-build the default suggested build method.

done in 0d75469

jessa0 commented:
feel free to leave comments on `script/container-build`

geogriff commented:
does this project prefer squashes, rebases, or merges?
